src/bits.rs

```diff
@@ -0,0 +1,723 @@
+                    }
+                    last_row = Some(row);
+                }
+                col += rle_len;
+            }
+        }
+        if first_row.is_some() {
+            let width = last_col.unwrap() - first_col.unwrap() + 1;
+            let height = last_row.unwrap() - first_row.unwrap() + 1;
+            Some(Region::new(first_col.unwrap() as isize, first_row.unwrap() as isize, width, height))
+        } else {
+            None
+        }
+    }
+
+    /// Copies `src` BitGrid into `dst` BitGrid, but fit into `dst_region` with clipping. If there
+    /// is a 10x10 pattern in source starting at 0,0 and the dst_region is only 8x8 with top left
+    /// corner at 3,3 then the resulting pattern will be clipped in a 2 pixel region on the bottom
+    /// and right and extend from 3,3 to 10,10. If `dst_region` extends beyond `dst`, the pattern
+    /// will be further clipped by the dimensions of `dst`.
+    ///
+    /// The bits are copied using an `|=` operation, so 1 bits in the destination will not ever be
+    /// cleared.
+    pub fn copy(src: &BitGrid, dst: &mut BitGrid, dst_region: Region) {
+        let dst_left   = cmp::max(0, dst_region.left()) as usize;
+        let dst_right  = cmp::min(dst.width() as isize - 1, dst_region.right()) as usize;
+        let dst_top    = cmp::max(0, dst_region.top()) as usize;
+        let dst_bottom = cmp::min(dst.height() as isize - 1, dst_region.bottom()) as usize;
+        if dst_left > dst_right || dst_top > dst_bottom {
+            // nothing to do because both dimensions aren't positive
+            return;
+        }
+
+        for src_row in 0..src.height() {
+            let dst_row = src_row + dst_top;
+            if dst_row > dst_bottom {
+                break;
+            }
+            let mut src_col = 0;   // word-aligned
+            while src_col < src.width() {
+                let dst_col = src_col + dst_left;    // not word-aligned
+                if dst_col > dst_right {
+                    break;
+                }
+                let dst_word_idx = dst_col / 64;
+                let shift = dst_col - dst_word_idx*64;  // right shift amount
+                let mut word = src[src_row][src_col/64];
+                // clear bits that would be beyond dst_right
+                if dst_right - dst_col + 1 < 64 {
+                    let mask = !((1u64 << (64 - (dst_right - dst_col + 1))) - 1);
+                    word &= mask;
+                }
+                dst[dst_row][dst_word_idx] |= word >> shift;
+                if shift > 0 && dst_word_idx+1 < dst.width_in_words() {
+                    dst[dst_row][dst_word_idx+1] |= word << (64 - shift);
+                }
+                src_col += 64;
+            }
+        }
+    }
+
+    /// Get a Region of the same size as the BitGrid.
+    pub fn region(&self) -> Region {
+        Region::new(0, 0, self.width(), self.height())
+    }
+
+    /// Clear this BitGrid.
+    pub fn clear(&mut self) {
```

Comment:
Why not iterate over all the components and just set to zero? 
Pseudo: `self[i][j] = 0 for i,j to width,height`

Reply:
Cool, sounds good, thanks for the suggestion.

Added bonus: I realized I was missing a unit test for this so I added that.